src/hooks/useNestedSafeOwners.tsx

```diff
@@ -0,0 +1,22 @@
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useMemo } from 'react'
+import useAllOwnedSafes from '@/components/welcome/MyAccounts/useAllOwnedSafes'
+import useWallet from './wallets/useWallet'
+
+export const useNestedSafeOwners = () => {
+  const { safe, safeLoaded } = useSafeInfo()
+  const wallet = useWallet()
+  const [allOwned] = useAllOwnedSafes(wallet?.address ?? '')
+  allOwned
+
+  const nestedSafeOwner = useMemo(() => {
+    if (!safeLoaded || !allOwned) return null
+
+    // Find an intersection of owned safes and the owners of the current safe
+    const ownerAddresses = safe?.owners.map((owner) => owner.value)
+
+    return allOwned[safe.chainId]?.filter((ownedSafe) => ownerAddresses?.includes(ownedSafe))
```

Comment:
As we're only concerned about the current chain, we should use the `[v1/chains/{chainId}/owners/{ownerAddress}/safes` endpoint instead of getting all the Safes across all chains.

Reply:
I also refactored the `useOwnedSafes` hook to use rtk query under the hood but kept the return type the same for now. Its something we should refactor as well though in a later iteration as the return type makes no sense currently.